### ISSUE TITLE
Searching for profiles

### ISSUE THREAD

**Issue by benjxia** [cse110-fa22-group25/warrantracker#32]
User story: As a landlord, I'd like the ability to quickly search through profiles by some sort of search bar.

**Comment by benjxia** [comment]
If we do a search, should it clear all tag filters? And then later we can apply tag filters on the search output.

**Comment by chrisliang12** [comment]
I think search should have a higher priority than tag. That means it will
ignore tag filter. If user first select some tag filter and do a search,
all profile matched the search results will show up , and only after user
clears the search text, tag filter can work again.

**Comment by benjxia** [comment]
What I have on the 32-searching-for-profiles branch so far is trying to mimic pcpartpicker.com filter behavior. (Incomplete and doesn't work).  If you do a search with filters already applied, it will search within the filters. If you do a search and then apply a filter, it will filter within the searched query (Backend can still be filter -> search, same behavior).

**Comment by benjxia** [comment]
ok so try to implement so that search will clear tag filters, but you can apply filter/tags on search results

**Comment by chrisliang12** [comment]
> ok so try to implement so that search will clear tag filters, but you can apply filter/tags on search results

That is a better solution. I will try to finish this tonight.

**Comment by benjxia** [comment]
you can nuke the current branch if you think it'd be easier to start from what it was

**Comment by benjxia** [comment]
can you see if the current version has any major bugs i didnt catch

**Comment by chrisliang12** [comment]
> can you see if the current version has any major bugs i didnt catch

I think there is no need to use a search btn. The functionality should be like the search_tag() function. When the text input changes, the content also changes. I will try to implement this later tonight.

**Comment by benjxia** [comment]
removed search button, content now changes with text

**Comment by chrisliang12** [comment]
Actually, if you want to implement this, I can leave this to you. ◔.̮◔✧

**Comment by benjxia** [comment]
ye its already done

**Comment by benjxia** [comment]
it was just change one word of an actionlistener

**Comment by chrisliang12** [comment]
Great!

**Comment by chrisliang12** [comment]
One small bug: search should not be case sensitive.

**Comment by chrisliang12** [comment]
> push a small change for search_tag() in your branch. The bug is fixed.

I mean the bug for issue #40

**Comment by benjxia** [comment]
fixed case sensitivity in search
